### PR TITLE
Display Owned Pokemon

### PR DESCRIPTION
**Pull request by jlora23**
Feel free to let me know if you want anything to be displayed differently! Take not that there are two ways to run this command. The first is to run `p!show` to display all owned Pokemon, and the second is to run `p!show <name of Pokemon>` to display a single pokemon.

### CODE REVIEW

**Comment by jlora23** [comment]
Feel free to let me know if you want anything to be displayed differently! Take not that there are two ways to run this command. The first is to run `p!show` to display all owned Pokemon, and the second is to run `p!show <name of Pokemon>` to display a single pokemon.

File: bot.py [hunk 1]
```diff
@@ -35,7 +35,7 @@ async def on_message(self, message: Message): # noqa: D102
 @bot.command()
 async def ping(ctx: commands.Context): # noqa: D103
 	await ctx.send('pong')
-	log.warning('log message')
+	#log.warning('log message')
```
Comment (dyc3) [inline bot.py:38]: Try to avoid unrelated changes

File: bot.py [hunk 2]
```diff
@@ -90,6 +90,30 @@ async def begin(ctx: commands.Context): # noqa: D103
...
 
 
+@bot.command(help='Add name to display specific Pokemon')
+async def show(ctx: commands.Context, single: Optional[str]): # noqa: D103
+	discord_id = ctx.author.id
+	base_msg = f"<@!{discord_id}> Here are all of your current Pokemon"
+	msg: Message = await ctx.send(base_msg)
+	#loads user class based upon their discord_id
+	user = await userprofile.load_profile(discord_id)
+	pokemon_list = [p async for p in user.pokemon_iter()]
```
Comment (dyc3) [inline bot.py:100]: No need to do this, because if the user only wants a single pokemon, this forces all the pokemon the user owns to get loaded from the database.

`pokemon_iter()` gives you an iterator that you can iterate over. You're just iterating over `pokemon_list` later, so instead of 

```python
for pokemon in pokemon_list:
```
try
```python
async for pokemon in user.pokemon_iter():
```

**Review by dyc3 — CHANGES_REQUESTED** [review]
Try not to blindly commit everything. You accidentally changed the commit that the submodule is pointing to.

Please also fix the linting errors, see the "Details" link below

**Comment by dyc3** [comment]
> You accidentally changed the commit that the submodule is pointing to.

I've fixed this for you. Please check out this branch again:
```
git checkout main
git branch -d display_pokemon
gh pr checkout 84
```

File: bot.py [hunk 3]
```diff
@@ -91,6 +91,29 @@ async def begin(ctx: commands.Context): # noqa: D103
 	await ctx.send("Profile created! `p!help` for more commands.")
 
 
+@bot.command(help='Add name to display specific Pokemon')
+async def show(ctx: commands.Context, single: Optional[str]): # noqa: D103
+	discord_id = ctx.author.id
+	base_msg = f"<@!{discord_id}> Here are all of your current Pokemon"
+	msg: Message = await ctx.send(base_msg)
+	#loads user class based upon their discord_id
+	user = await userprofile.load_profile(discord_id)
```
Comment (dyc3) [inline bot.py:100]: The type checker is yelling at you for this. To fix it, add the following line below this one:
```python
assert isinstance(user, UserProfile)
```

**Review by dyc3 — CHANGES_REQUESTED** [review]

File: bot.py (hunk 3 above)
Comment (dyc3) [reply bot.py:100]: and you should remove that comment, because it states the obvious

**Review by dyc3 — APPROVED** [review]
